x-pack/test/observability_ai_assistant_api_integration/tests/public_complete/public_complete.spec.ts

```diff
@@ -209,6 +241,75 @@ export default function ApiTest({ getService }: FtrProviderContext) {
         expect(request.messages[0].content).to.contain('This is a random instruction');
       });
     });
+
+    describe('with openai format', async () => {
+      let responseBody: string;
+
+      before(async () => {
+        responseBody = await getOpenAIResponse(async (conversationSimulator) => {
+          await conversationSimulator.next('Hello');
+          await conversationSimulator.complete();
+        });
+      });
+
+      function extractDataParts(lines: string[]) {
+        return lines.map((line) => {
+          // .replace is easier, but we want to verify here whether
+          // it matches the SSE syntax (`data: ...`)
+          const [, dataPart] = line.match(/^data: (.*)$/) || ['', ''];
+          return dataPart.trim();
+        });
+      }
+
+      it('outputs each line an SSE-compatible format', () => {
+        const lines = responseBody.split('\n\n').filter(Boolean);
+
+        lines.forEach((line) => {
+          expect(line.match(/^data: /));
+        });
+
+        const dataParts = extractDataParts(lines);
+
+        expect(dataParts.length).to.be(2);
+
+        dataParts.every((part) => {
+          expect(part).not.to.be.empty();
+        });
+
+        expect(last(dataParts)).to.be('[DONE]');
+
+        expect(() => {
+          dataParts.slice(0, -1).forEach((part) => {
+            JSON.parse(part);
+          });
+        }).not.to.throwException();
```

Comment:
I'm not too familiar with SSE response format. My understanding is that each line must start with `data:` like

```
data: First line
data: Second line
data: Third line
```

It's not entirely clear to me whether the rest is is SSE format or something custom we require - like `[DONE]` as the last event.

Reply:
`[DONE]` I think is specific to OpenAI, do you mean we should move that to a different test to make that more clear (in that it is not SSE-specific)?